integration-tests/src/lib.rs

```diff
@@ -431,6 +450,105 @@ fn reserve_transfer_native_asset_from_para_to_system_para() {
 	assert!(receiver_balance_after < receiver_balance_before + amount_to_send);
 }
 
+#[test]
+fn place_coretime_spot_order_from_para_to_relay() {
+	init_tracing();
+
+	let beneficiary: sp_runtime::AccountId32 = [1u8; 32].into();
+
+	// Setup: reserve transfer from relay to Pop, so that sovereign account accurate for return transfer
+	let amount_to_send: Balance = pop_runtime::UNIT * 1000;
+	fund_pop_network(RococoRelaySender::get(), amount_to_send, beneficiary.clone());
+
+	let message = {
+		let assets: Asset = (Here, 10 * pop_runtime::UNIT).into();
+		let beneficiary = AccountId32 { id: beneficiary.clone().into(), network: None }.into();
+		let spot_order = <RococoRelay as Chain>::RuntimeCall::OnDemandAssignmentProvider(
+			assigner_on_demand::Call::<<RococoRelay as Chain>::Runtime>::place_order_keep_alive {
+				max_amount: 1 * pop_runtime::UNIT,
+				para_id: AssetHubRococoPara::para_id().into(),
+			},
+		);
+
+		// Set up transact status response handler
+		let query_id = PopNetworkPara::execute_with(|| {
+			<PopNetwork<RococoMockNet> as PopNetworkParaPallet>::PolkadotXcm::new_query(
+				PopNetworkPara::parent_location(),
+				10u32.into(),
+				Location::here(),
+			)
+		});
+
+		let message = Xcm::builder()
+			.withdraw_asset(assets.clone().into())
+			.buy_execution(assets.clone().into(), Unlimited)
+			.transact(
+				OriginKind::SovereignAccount,
+				Weight::from_parts(220_000_000, 15_000),
+				spot_order.encode().into(),
+			)
+			.report_transact_status(QueryResponseInfo {
+				destination: RococoRelay::child_location_of(PopNetworkPara::para_id()),
+				query_id,
+				max_weight: Weight::from_parts(250_000_000, 10_000),
+			})
+			.refund_surplus()
+			.deposit_asset(assets.into(), beneficiary)
+			.build();
+		message
+	};
+
+	let destination = PopNetworkPara::parent_location().into_versioned();
+	PopNetworkPara::execute_with(|| {
+		let res = <PopNetworkPara as Chain>::RuntimeCall::PolkadotXcm(pallet_xcm::Call::<
+			<PopNetworkPara as Chain>::Runtime,
+		>::send {
+			dest: bx!(destination),
+			message: bx!(VersionedXcm::V4(message)),
+		})
+		// TODO: replace root with signed, currently prohibited by HashedDescription<AccountId, DescribeFamily<DescribeBodyTerminal>> (https://github.com/paritytech/polkadot-sdk/blob/a6713c55fd5082d333518c3ca13f2a4294726fcc/polkadot/runtime/rococo/src/xcm_config.rs#L67) rather than HashedDescription<AccountId, DescribeFamily<DescribeAllTerminal>> (https://github.com/polkadot-fellows/runtimes/blob/e42821da8d85f721d0dd1670dfb23f4dd91bd3e8/relay/kusama/src/xcm_config.rs#L76)
+		//.dispatch(RawOrigin::Signed(beneficiary).into());
+		.dispatch(RawOrigin::Root.into());
+
+		assert!(res.is_ok());
+		type RuntimeEvent = <PopNetworkPara as Chain>::RuntimeEvent;
+		// Check that the message was sent
+		assert_expected_events!(
```

Comment:
Should this be added to the `PopNetworkPara::execute_with` that is handling events already?

Reply:
Sorry, just saw you addressed these, thank you!